.github/workflows/e2e.yml

```diff
@@ -0,0 +1,123 @@
+#  the License.
+
+# This workflow will build a Java project with Maven
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+# Note: Any changes to this workflow would be used only after merging into develop
+name: Build e2e tests
+
+on:
+  push:
+    branches: [ develop ]
+  pull_request:
+    branches: [ develop ]
+    types: [opened, synchronize, reopened, labeled]
+  workflow_dispatch:
+
+jobs:
+  modules:
+    runs-on: ubuntu-latest
+    # We allow builds:
+    # 1) When triggered manually
+    # 2) When it's a merge into a branch
+    # 3) For PRs that are labeled as build and
+    #  - It's a code change
+    #  - A build label was just added
+    # A bit complex, but prevents builds when other labels are manipulated
+    if: >
+      github.event_name == 'workflow_dispatch'
+      || github.event_name == 'push'
+      || (contains(github.event.pull_request.labels.*.name, 'build')
+          && (github.event.action != 'labeled' || github.event.label.name == 'build')
+          )
+    steps:
+      - uses: actions/checkout@v2.3.4
+        if: github.event_name != 'workflow_dispatch'
+        with:
+          path: plugin
+      - uses: dorny/paths-filter@b2feaf19c27470162a626bd6fa8438ae5b263721
+        if: github.event_name != 'workflow_dispatch'
+        id: filter
+        with:
+          working-directory: plugin
+          filters: |
+            core-plugins:
+              - 'core-plugins/**'
+      - name: Set modules list
+        id: sml
+        run: |
+          if [ ${{ github.event_name }} != 'workflow_dispatch' ]; then
+            echo "::set-output name=matrix::$( jq '.' --compact-output <<< '${{ steps.filter.outputs.changes }}')"
+          else
+            echo "::set-output name=matrix::['core-plugins']"
+          fi
+        shell: bash
+    outputs:
+      matrix: ${{ steps.sml.outputs.matrix }}
+
+  build:
+    needs: modules
+    runs-on: k8s-runner-e2e
+    if: needs.modules.outputs.matrix != '[]' && needs.modules.outputs.matrix != ''
+    strategy:
+      matrix:
+        module: ${{ fromJson(needs.modules.outputs.matrix) }}
+    steps:
+      # Pinned 1.0.0 version
+      - uses: actions/checkout@v2.3.4
+        with:
+          path: plugin
+          submodules: 'recursive'
+      - name: Checkout e2e test repo
+        uses: actions/checkout@v2.3.4
+        with:
+          repository: cdapio/cdap-e2e-tests
+          path: e2e
+      - name: Cache
+        uses: actions/cache@v2.1.3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ github.workflow }}-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-${{ github.workflow }}
+      - name: Run required e2e tests
+        if: github.event_name != 'workflow_dispatch' && (!contains( github.event.head_commit.message, 'Merge pull request'))
```

Comment:
instead of `commit.message`,use `github.event_name == on_push`.

Reply:
The case which you described the event for that should be `synchronize` not `push`. Could you please test it once?

Reply:
Yes correct, that is `synchronize` event.
Updated condition.

Sorry I was confused  as commit gets pushed to branch from which PR is raised and on that branch's workflow it triggers push event.